calicoctl/calico_ctl/checksystem.py

```diff
@@ -246,3 +253,37 @@ def _check_etcd_version():
             system_ok = False
 
     return system_ok
+
+
+def _check_kernel_version():
+    """
+    Check that the kernel version meets the minimum required version for Calico.
+    Valid kernel versions have 1 Kernel digit, 1-2 Major digits, and optionally
+    1-2 Minor digits for versioning.  The full version can be followed by at
+    least one non-digit value, then any other characters:
+    Ex.    "2.24.32-100-generic"
+           "3.19.0a-dev"
+           "2.8.1"
+           "3.0"
+           "4.5-rc3"
+    """
+    full_version = platform.uname()[2]
+
+    # Make sure version number makes sense and captures version numbers
+    regex = re.compile("^([\d]\.[\d]{1,2}(\.[\d]{1,2})?)(?:\D.*)?$")
```

Comment:
Wouldn't the first bit be easier as something like `([\d](\.[\d]?){1,3}`

i.e. capture the first number and between 1-3  .x 's